src/Engine/GameInstance.java

```diff
@@ -255,7 +260,8 @@ public static GameInstance loadSave(String filename)
       verInfo = (GameVersion) in.readObject();
       if( !new GameVersion().isEqual(verInfo) )
       {
-        throw new Exception(String.format("Deserializing failed! Save is incompatible version: %s", verInfo.toString()));
+        throw new Exception(String.format("Deserializing failed! Save is incompatible version: %s",
+            (null == verInfo) ? "unknown" : verInfo.toString()));
```

Comment:
The fact that you have to put this same logic in two places makes me think there's a better way to organize this. I guess this is so you can do the check both when looking for save files and when attempting to load them?

It's also a little odd to have a function named `isSaveCompatible()` and then check compatibility in `loadSave()` without calling that function.

Reply:
...hah, yeah.
It's a little ugly, for sure, but I don't really want to make a file stream an argument of the function.
I guess I could make overloads of isSaveCompatible, one of which takes a path, and calls the other with the stream.

...then again, that sounds like complicated control flow. Maybe the correct solution is to just delete that logic from the reading function. We'll already have marked the save as incompatible, and we're safe from exceptions, so maybe we should just try what the user asks.

Reply:
Yeah, I'd be fine with that approach. If the versions don't match, print a warning, and just try to load it anyway. Catch the exception of course, so it's clear what happened, and they can pick a different one.